apps/judicial-system/web/src/api/index.ts

```diff
@@ -1,30 +1,86 @@
 import 'isomorphic-fetch'
-import { Case } from '../types'
+import { Case, CreateCaseRequest } from '../types'
 
-interface GetCasesResponse {
-  ok: boolean
-  code: number
-  cases?: Case[]
-  message?: string
-}
+// const getCaseById: (caseId: string) => Promise<GetCaseByIdResponse> = async (
+//   caseId: string,
+// ) => {
+//   const response = await fetch(`/api/case/${caseId}`)
+
+//   if (response.ok) {
+//     const theCase: Case = await response.json()
+//     return {
+//       httpStatusCode: response.status,
+//       case: theCase,
+//     }
+//   } else {
+//     return {
+//       httpStatusCode: response.status,
+//     }
+//   }
+// }
 
-export const getCases: () => Promise<GetCasesResponse> = async () => {
-  const response = await fetch('/api/cases')
+export const getCases: () => Promise<Case[]> = async () => {
+  try {
+    const response = await fetch('/api/cases')
 
-  if (response.ok) {
-    const cases = await response.json()
-    return {
-      ok: true,
-      code: response.status,
-      cases,
+    if (response.ok) {
+      const cases = await response.json()
+      return cases
+    } else {
+      throw new Error(response.statusText)
     }
-  } else {
+  } catch (ex) {
     // TODO: Log error
+    console.log(ex)
+  }
+}
+
+export const createCase: (
+  caseToCreate: CreateCaseRequest,
+) => Promise<string> = async (caseToCreate: CreateCaseRequest) => {
+  try {
+    const response = await fetch('/api/case', {
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(caseToCreate),
+    })
+
+    if (response.ok) {
+      const responseJSON: Case = await response.json()
+      return responseJSON.id
+    } else {
+      throw new Error(response.statusText)
+    }
+  } catch (ex) {
+    // TODO: log error
+    console.log(ex)
+  }
+}
+
+export const saveCase: (
+  caseId: string,
+  caseField: string,
+  caseFieldValue: string,
+) => Promise<number> = async (
+  caseId: string,
+  caseField: string,
+  caseFieldValue: string,
+) => {
+  if (caseId !== '') {
+    const propertyChange = JSON.parse(`{"${caseField}": "${caseFieldValue}"}`)
+    const response = await fetch(`/api/case/${caseId}`, {
```

Comment:
Will you never be saving multiple fields at the same time?
For example, when previous saves failed?

Reply:
It depends on if we are going to opt for the solution to display an error message at the top of the screen, saying that something went wrong, try again. In that solution we'd collect all the fields that were not saved and save them together. If we opt for the "this specific input did not save approach" with a checkmark icon for inputs that are saved and an error icon for those that didn't, we might want to save just that field. Did we decide to implement the first solution when we talked about this earlyer?

Reply:
No, we did not decide on anything.